lib/pbench/server/api/resources/upload_api.py

```diff
@@ -128,35 +159,51 @@ def put(self, controller):
             )
             dataset.add()
         except DatasetDuplicate:
-            self.logger.info("Dataset already exists {}", filename)
+            self.logger.info(
+                "Dataset already exists, user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             response = jsonify(dict(message="Dataset already exists"))
             response.status_code = 200
             return response
         except Exception:
-            self.logger.exception("unable to create dataset for {}", filename)
+            self.logger.exception(
+                "unable to create dataset for user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
         if tar_full_path.is_file() or md5_full_path.is_file():
             self.logger.error(
-                "Dataset or corresponding md5 file already present on the disc, {}",
-                filename,
+                "Dataset or corresponding md5 file already present on the disc, {}, {}",
+                tar_full_path,
+                md5_full_path,
             )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
-        self.logger.info("Uploading file {} to {}", filename, dataset)
+        self.logger.info(
+            "Uploading file {} (user = {}, ctrl = {}) to {}",
+            username,
+            controller,
+            filename,
+            dataset,
+        )
```

Comment:
I think these parameters are mis-ordered.

Reply:
Yes, along with a number of other errors with logged output!